### PR TITLE
Fix: `brace-style` doesn't report some closing brace errors

### LINKED ISSUE #3486: "allman" option of `brace-style` rule does not perform as documented
According to the documentation, these should cause an error:

```js
/*eslint brace-style: [2, "allman"]*/

if (foo)
{
  bar(); }

try
{
  somethingRisky();
} catch(e)
{
  handleError();
}
```

Using current master, I get no errors.

### PR DESCRIPTION
**Pull request by BYK**
Fixes #3486

### CODE REVIEW

File: tests/lib/rules/brace-style.js [hunk 1]
```diff
@@ -110,19 +110,46 @@ ruleTester.run("brace-style", rule, {
...
         { code: "if (foo)\n{ poop();\n} \nelse if (bar) {\nbaz();\n} else if (thing) {\nboom();\n}\nelse {\nqux();\n}", options: ["allman"], errors: [
             { message: BODY_MESSAGE, type: "ExpressionStatement" },
             { message: OPEN_MESSAGE_ALLMAN, type: "IfStatement" },
+            { message: CLOSE_MESSAGE_STROUSTRUP_ALLMAN, type: "IfStatement" },
             { message: OPEN_MESSAGE_ALLMAN, type: "IfStatement" },
             { message: OPEN_MESSAGE_ALLMAN, type: "IfStatement" }
         ] },
+        {
+            code: [
+                "if (foo)",
```
Comment (ilyavolodin) [inline tests/lib/rules/brace-style.js]: I would go with the same style that is already used above (single string with "/n"). In general, we should just use multiline strings for those, I think

**Comment by BYK** [comment]
@ilyavolodin  - done. I find this style very unreadable though and agree with multi-line strings. Not sure if they are available on node 0.10 though.

**Comment by IanVS** [comment]
I had to write a little script to run through all the strings in `newline-after-var` just so I could get a handle on what was being tested.  Would have been much easier if they were multi-line.

**Comment by BYK** [comment]
@IanVS - Yeah, looks like node 0.10 doesn't support multi-line template strings even with `--harmony` :disappointed:

**Comment by BYK** [comment]
Is this good to go?

**Comment by ilyavolodin** [comment]
LGTM

**Pull request merged**